### DISCUSSION TITLE
updating charging profile window?

### DISCUSSION THREAD

**Discussion by dougstarfish** [bimmerconnected/bimmer_connected#577] [General]
Does anybody know how to update the charging window? 

Running this is easy enough: result = asyncio.run(vehicle.remote_services.trigger_charging_profile_update(charging_mode= 'DELAYED_CHARGING')

But I haven't been able to figure out how to set the window. None of the methods under remote_services seem to take an object that corresponds to a profile/window.

**Comment by rikroe** [comment]
Currently, setting the window is not possible. I don't have an EV so cannot test it and I am not sure if we would want some kind of validation on the payload (which adds even more complexity).

We could start by adding `kwargs` to `trigger_charging_profile_update` so that it is possible to add arbitrary data, however I don't know how the API will react.

**Reply by dougstarfish** [reply]
Okay, I was wondering if it was just a missing part of trigger_charging_profile_update given what I've seen online, I was also thinking one could get the current charging profile and just modify it and with the right REST change-o, upload a new version.

**Comment by swestland85** [comment]
Has there any way found to update the charging windows through HA?
